nvflare/fuel/utils/class_utils.py

```diff
@@ -136,3 +136,20 @@ def get_module_name(self, class_name) -> Optional[str]:
             The module name if found.
         """
         return self._class_table.get(class_name, None)
+
+
+def _retrieve_parameters(class__, parameters):
+    constructor = class__.__init__
+    constructor__parameters = inspect.signature(constructor).parameters
+    parameters.update(constructor__parameters)
+    if "args" in constructor__parameters.keys() and "kwargs" in constructor__parameters.keys():
+        for item in class__.__bases__:
+            parameters.update(_retrieve_parameters(item, parameters))
+    return parameters
+
+
+def get_component_init_parameters(component):
```

Comment:
Please add unit test cases for simple and complex (subclassing) object structures.

Reply:
Added docstring and unit test.